flipt-client-go/evaluation_test.go

```diff
@@ -30,6 +34,11 @@ func init() {
 	}
 }
 
+func TestInvalidAuthentication(t *testing.T) {
+	_, err := flipt.NewClient(flipt.WithURL(fliptUrl), flipt.WithClientTokenAuthentication("invalid"))
+	assert.EqualError(t, err, "invalid authentication")
```

Comment:
i know this isnt the correct error string, but i just wanted to see if there is any error

Reply:
we need to do some `variant, err := evaluationClient.EvaluateVariant(...)` to get the error.

Reply:
it should return the error on the initial pulling of the state I would think. I updated the test and still no error, maybe we arent returning it correctly from the engine?

Reply:
let me give it a try.